generators/new_collateral/config/parser.go

```diff
@@ -0,0 +1,85 @@
+package config
+
+import (
+	"errors"
+
+	"github.com/BurntSushi/toml"
+	"github.com/makerdao/vdb-mcd-transformers/generators/new_collateral/helpers"
+	"github.com/makerdao/vdb-mcd-transformers/generators/new_collateral/types"
+	"github.com/mitchellh/mapstructure"
+)
+
+type IParse interface {
+	ParseCurrentConfig(configFilePath, configFileName string) (types.TransformersConfig, error)
+}
+type Parser struct{}
+
+func (Parser) ParseCurrentConfig(configFilePath, configFileName string) (types.TransformersConfig, error) {
+	var tomlConfig types.TransformersConfigForToml
+	fullConfigFilePath := helpers.GetFullConfigFilePath(configFilePath, configFileName)
+	_, decodeErr := toml.DecodeFile(fullConfigFilePath, &tomlConfig)
+	if decodeErr != nil {
+		return types.TransformersConfig{}, decodeErr
+	}
+
+	metadata, metadataErr := parseExporterMetaData(tomlConfig)
+	if metadataErr != nil {
+		return types.TransformersConfig{}, metadataErr
+	}
+
+	transformerExporters, transformerExportersErr := parseTransformerExporters(tomlConfig)
+	if transformerExportersErr != nil {
+		return types.TransformersConfig{}, transformerExportersErr
+	}
+
+	return types.TransformersConfig{
+		ExporterMetadata:     metadata,
+		TransformerExporters: transformerExporters,
+		Contracts:            tomlConfig.Contracts,
+	}, nil
+}
+
+func parseExporterMetaData(tomlConfig types.TransformersConfigForToml) (types.ExporterMetaData, error) {
+	home, homeOk := tomlConfig.Exporter["home"].(string)
+	name, nameOk := tomlConfig.Exporter["name"].(string)
+	save, saveOk := tomlConfig.Exporter["save"].(bool)
+	schema, schemaOk := tomlConfig.Exporter["schema"].(string)
+	if !homeOk || !nameOk || !saveOk || !schemaOk {
+		return types.ExporterMetaData{}, errors.New("error asserting exporter meta data types")
```

Comment:
also super minor but I wonder if it'd be helpful to format in the various `ok` values so that we can see what's missing in the error

Reply:
addressed in 0018720